### ISSUE TITLE
Error with distinct generic TableRef

### ISSUE THREAD

**Issue by mathsaey** [nim-lang/Nim#6060]
The following:

```
import tables

type MyTab[A,B] = distinct TableRef[A,B]

proc `$`[A,B](t: MyTab[A,B]): string =
  "My special table " & $TableRef[A,B](t)

proc create[A,B](): MyTab[A,B] = MyTab(newTable[A,B]())

var a = create[int,int]()
echo a
```

Does not compile on the current devel:

```
test.nim(11, 6) template/generic instantiation from here
test.nim(6, 39) Error: type mismatch: got (MyTab[system.int, system.int]) but expected 'TableRef[system.int, system.int]'
```

Curiously, it does seem to work when `Table` is used instead of `TableRef`

**Comment by ringabout** [comment]
It has been fixed since Nim 1.0
output:
```
"My special table {:}"
```

**Comment by Clyybber** [comment]
Closed with #15366
